slack_connector/main.py

```diff
@@ -78,9 +106,23 @@ def filter_data(df):
         A dataframe with processed columns
     """
 
+    # Filter dataframe to keep required columns
     df = df[['type','text','user','ts','client_msg_id','subtype']]
-    #df.to_csv('slack.csv')
-    #df.to_csv(args.local_dir+'/'+args.file_name, index=False)
+
+    # Raise error if datframe is empty
+    if df.empty == True:
+        raise NoMessageError()
```

Comment:
This will check will not be required if code is rearranged as per my comments above?